src/sys/unix/net.rs

```diff
@@ -13,36 +13,26 @@ pub fn new_ip_socket(addr: SocketAddr, socket_type: libc::c_int) -> io::Result<l
 
 /// Create a new non-blocking socket.
 pub fn new_socket(domain: libc::c_int, socket_type: libc::c_int) -> io::Result<libc::c_int> {
-    #[cfg(any(
-        target_os = "android",
-        target_os = "bitrig",
-        target_os = "dragonfly",
-        target_os = "freebsd",
-        target_os = "linux",
-        target_os = "netbsd",
-        target_os = "openbsd"
-    ))]
-    let socket_type = socket_type | libc::SOCK_NONBLOCK | libc::SOCK_CLOEXEC;
-
-    // Gives a warning for platforms without SOCK_NONBLOCK.
-    #[allow(clippy::let_and_return)]
-    let socket = syscall!(socket(domain, socket_type, 0));
+    #[cfg(not(any(target_os = "ios", target_os = "macos", target_os = "solaris")))]
```

Comment:
I don't think we want to do cfg this way. If we add a new target in the old code it would explode indicating that we need to look if this would work on the new platform, for the new code this is not the case.